In [1]:
import pandas as pd

### Loading the dataset

In [2]:
df = pd.read_csv("Mall_Customers.csv")

<p>Check first 5 records

In [3]:
df.head()

CustomerID  Gender  Age  Annual Income (k$)  Spending Score (1-100)
0           1    Male   19                  15                      39
1           2    Male   21                  15                      81
2           3  Female   20                  16                       6
3           4  Female   23                  16                      77
4           5  Female   31                  17                      40

<p>Perform basics analysis

In [4]:
df.describe()

CustomerID         Age  Annual Income (k$)  Spending Score (1-100)
count  200.000000  200.000000          200.000000              200.000000
mean   100.500000   38.850000           60.560000               50.200000
std     57.879185   13.969007           26.264721               25.823522
min      1.000000   18.000000           15.000000                1.000000
25%     50.750000   28.750000           41.500000               34.750000
50%    100.500000   36.000000           61.500000               50.000000
75%    150.250000   49.000000           78.000000               73.000000
max    200.000000   70.000000          137.000000               99.000000

### Understand Customer Demographics

<p> Calculate the average Age, Annual Income, and Spending Score to understand the typical customer profile.

In [5]:
df.loc[:, ~df.columns.isin(["CustomerID", "Gender"])].mean(numeric_only=True)  # Select all numeric column

Age                       38.85
Annual Income (k$)        60.56
Spending Score (1-100)    50.20
dtype: float64

In [6]:
df[["Age", "Annual Income (k$)", "Spending Score (1-100)"]].mean()             # Select intended column    

Age                       38.85
Annual Income (k$)        60.56
Spending Score (1-100)    50.20
dtype: float64

> Average age 38.85 year old suggests that the mall's typical customer base consists of middle-aged adults.<br>
> Average income of $60.56k suggests that customers are generally from a moderate-income demographic.<br>
> Average spending score 50.2 shows that spending behavior is fairly balanced, as the score is near the midpoint of 1–100

<p>Analyze customer distribution by gender. Does one gender tend to spend more or earn more on average?</p>

In [7]:
gender_distribution = pd.DataFrame([df['Gender'].value_counts(), df['Gender'].value_counts(normalize=True) * 100],
                                   index=["Count", "Distribution (%)"])
gender_distribution.transpose()


Count  Distribution (%)
Female  112.0              56.0
Male     88.0              44.0

In [8]:
#income_distribution = pd.DataFrame(df['Annual Income (k$)'].mean())
df['Annual Income (k$)'].mean()

60.56